server/src/main/java/io/crate/exceptions/JobKilledException.java

```diff
@@ -21,30 +21,44 @@
 
 package io.crate.exceptions;
 
+import io.crate.data.Killable;
 import org.elasticsearch.ElasticsearchException;
 import org.elasticsearch.common.io.stream.StreamInput;
 
+import javax.annotation.Nonnull;
 import javax.annotation.Nullable;
 import java.io.IOException;
 
 public class JobKilledException extends ElasticsearchException implements UnscopedException {
 
     public static final String MESSAGE = "Job killed";
+    public final Killable.Type killType;
 
-    public static JobKilledException of(@Nullable String reason) {
-        return reason == null ? new JobKilledException() : new JobKilledException(reason);
+    public static JobKilledException of(@Nullable String reason, @Nonnull Killable.Type killType) {
+        return reason == null ? new JobKilledException(killType) : new JobKilledException(reason, killType);
+    }
+
+    public static JobKilledException ofHard(@Nullable String reason) {
+        return reason == null ? new JobKilledException(Killable.Type.HARD) : new JobKilledException(reason, Killable.Type.HARD);
+    }
+
+    public static JobKilledException ofSoft(@Nullable String reason) {
+        return reason == null ? new JobKilledException(Killable.Type.SOFT) : new JobKilledException(reason, Killable.Type.SOFT);
     }
 
     public JobKilledException(final StreamInput in) throws IOException {
         super(in);
+        killType = Killable.Type.HARD;  // fix this
     }
 
-    private JobKilledException(String reason) {
+    protected JobKilledException(String reason, Killable.Type killType) {
         super("Job killed. " + reason);
+        this.killType = killType;
     }
 
-    private JobKilledException() {
+    protected JobKilledException(Killable.Type killType) {
         super(MESSAGE);
+        this.killType = killType;
```

Comment:
Why the permission change if there are also static factory methods?

Reply:
sorry for confusion, I tried extending it initially and not fully reverted afterwards. fixing now